packages/producer-key-readmodel-writer-sql/src/producerKeyConsumerServiceV2.ts

```diff
@@ -0,0 +1,72 @@
+import { keyToProducerJWKKey } from "pagopa-interop-commons";
+import {
+  AuthorizationEventEnvelopeV2,
+  fromProducerKeychainV2,
+  genericInternalError,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { CustomReadModelService } from "./readModelService.js";
+
+export async function handleMessageV2(
+  message: AuthorizationEventEnvelopeV2,
+  producerJWKKeyReadModelService: CustomReadModelService
+): Promise<void> {
+  await match(message)
+    .with({ type: "ProducerKeychainKeyAdded" }, async (message) => {
+      const producerKeychain = message.data.producerKeychain
+        ? fromProducerKeychainV2(message.data.producerKeychain)
+        : undefined;
+      const key = producerKeychain?.keys.find(
+        (key) => key.kid === message.data.kid
+      );
+      if (!key) {
+        throw Error(
```

Comment:
Usually, we try to not throw an `Error` directly but to use a function that we define and control, in these cases I think we use `genericInternalError` (same for all other throws in the file)

```suggestion
        throw genericInternalError(
```